libs/dh/reports/feature-settlement-reports/src/components/request-report/request-report-modal.component.ts

```diff
@@ -180,39 +178,48 @@ export class DhRequestReportModal extends WattTypedModal<SettlementReportRequest
     allowLargeTextFiles: new FormControl<boolean>(false, { nonNullable: true }),
   });
 
-  showEnergySupplierDropdown$ = of(this.modalData.isFas).pipe(
-    map((isFas) => isFas || this.modalData.marketRole === EicFunction.SystemOperator),
-    tap((showEnergySupplierDropdown) => {
-      if (showEnergySupplierDropdown) {
-        this.form.addControl(
-          'energySupplier',
-          new FormControl<string | null>(ALL_ENERGY_SUPPLIERS, Validators.required)
-        );
-      }
-    })
+  showEnergySupplierDropdown = toSignal(
+    of(this.modalData.isFas).pipe(
+      map((isFas) => isFas || this.modalData.marketRole === EicFunction.SystemOperator),
+      tap((showEnergySupplierDropdown) => {
+        if (showEnergySupplierDropdown) {
+          this.form.addControl(
+            'energySupplier',
+            new FormControl<string | null>(ALL_ENERGY_SUPPLIERS, Validators.required)
+          );
+        }
+      })
+    ),
+    { initialValue: false }
```

Comment:
comment: what is this cursed code 😄 I know it is not your doing, so I'll let it slide. But this should really just be `showEnergySupplierDropdown = this.modalData.isFas`, with the side effect being moved to the constructor.